alloptions-limit.php

```diff
@@ -152,41 +145,37 @@ function wpcom_vip_sanity_check_alloptions_notify( $size, $blocked = false ) {
 			size_format( $size )
 		);
 
-		$to_irc = $subject . ' #vipoptions';
-
 		// Send to IRC, if we have a host configured
-		if ( defined( 'ALERT_SERVICE_ADDRESS' ) && ALERT_SERVICE_ADDRESS ) {
-			if ( 'production' === $environment ) {
-
-				if ( ! wpcom_vip_alloptions_size_is_acked() ) {
-					wpcom_vip_irc( '#vip-deploy-on-call', $to_irc, $irc_alert_level, 'a8c-alloptions' );
-				}
-
-				// Send to OpsGenie
-				$alerts = Alerts::instance();
-				$alerts->opsgenie(
-					$subject,
-					array(
-						'alias'       => 'alloptions/' . $site_id,
-						'description' => 'The size of AllOptions is reaching the max limit of 1 MB.',
-						'entity'      => (string) $site_id,
-						'priority'    => $opsgenie_alert_level,
-						'source'      => 'sites/alloptions-size',
-					),
-					'alloptions-size-alert',
-					'10'
-				);
-
-			}
-		}
+		if (
+			defined( 'ALERT_SERVICE_ADDRESS' ) &&
+			ALERT_SERVICE_ADDRESS &&
+			'production' === $environment &&
+			$really_blocked
+		) {
+
+			// Send to OpsGenie
+			$alerts = Alerts::instance();
+			$alerts->opsgenie(
+				$subject,
+				array(
+					'alias'       => 'alloptions/' . $site_id,
+					'description' => sprintf( 'The size of AllOptions has breached %s bytes', VIP_ALLOPTIONS_ERROR_THRESHOLD ),
+					'entity'      => (string) $site_id,
+					'priority'    => 'P3',
+					'source'      => 'sites/alloptions-size',
+				),
+				'alloptions-size-alert',
+				'10'
+			);
 
-		$email_recipient = defined( 'VIP_ALLOPTIONS_NOTIFY_EMAIL' ) ? VIP_ALLOPTIONS_NOTIFY_EMAIL : false;
+		}
 
-		if ( $email_recipient ) {
-			$size = size_format( $size );
+		/**
+		 * Fires under alloptions warning conditions
+		 *
+		 * @param bool $really_blocked False if alloptions size is large. True if site loading is being blocked.
+		 */
+		do_action( 'vip_alloptions_notification', $really_blocked );
```

Comment:
Let's move this hook up higher before "Send notices to VIP staff". I don't think it needs to be under any conditionals aside from being under the rate-limiting ones.

Then can short circuit all the "Send notices to VIP staff" section conditionals and early abort if opsgenie won't be needed.